testsuite/src/main/java/io/netty5/testsuite/transport/socket/SocketTestPermutation.java

```diff
@@ -203,18 +204,39 @@ public List<BootstrapFactory<Bootstrap>> datagramSocket() {
                 () -> new Bootstrap().group(nioWorkerGroup).channel(NioDatagramChannel.class)
         );
     }
+
     public static DomainSocketAddress newDomainSocketAddress() {
-        try {
-            File file;
-            do {
-                file = PlatformDependent.createTempFile("NETTY", "UDS", null);
-                if (!file.delete()) {
-                    throw new IOException("failed to delete: " + file);
-                }
-            } while (file.getAbsolutePath().length() > 128);
-            return new DomainSocketAddress(file);
-        } catch (IOException e) {
-            throw new IllegalStateException(e);
+        return new DomainSocketAddress(DomainSocketDirectory.nextSocketName());
+    }
+
+    /**
+     * Separate class so we only initialize this field if we really need it.
+     */
+    private static final class DomainSocketDirectory {
+        public static final int MAX_PATH_LEN = 128;
+        public static final AtomicInteger COUNTER = new AtomicInteger();
+        public static final File DIR;
+
+        static {
+            try {
+                int maxLen = MAX_PATH_LEN - File.separator.length() - 2 * Integer.BYTES;
+                File file = null;
+                do {
+                    if (file != null && !file.delete()) {
+                        throw new IOException("failed to delete: " + file);
+                    }
+                    file = PlatformDependent.createTempDirectory("NETTY-UDS", null).getAbsoluteFile();
+                } while (file.toString().length() > maxLen);
```

Comment:
Let's ensure we not loop forever
```suggestion
                int maxTries = 64;
                do {
                    if (maxTries-- == 0) {
                        throw new ExceptionInInitializerError("Unable to create temporary file with path length <= " + maxLen);
                    }
                    if (file != null && !file.delete()) {
                        throw new IOException("failed to delete: " + file);
                    }
                    file = PlatformDependent.createTempDirectory("NETTY-UDS", null).getAbsoluteFile();
                } while (file.toString().length() > maxLen);
```